code/game/objects/machinery/mortar.dm

```diff
@@ -253,6 +277,20 @@
 	qdel(mortar_shell)
 	firing = FALSE
 
+/obj/machinery/deployable/mortar/attack_ai(mob/living/silicon/ai/user)
+	if(user.linked_artillery == src)
+		switch(tgui_alert(usr, "This artillery piece is linked to you. Do you want to unlink yourself from it?", "Artillery Targeting", list("Yes", "No")))
```

Comment:
You can do the same thing with a single if. Including the user.linked artillery == src